### ISSUE TITLE
4.1.3 hangs on "Connected, checking location info."

### ISSUE THREAD

**Issue by aneng64** [Zomboided/service.vpn.manager#101]
Hi... me again!  I installed 4.1.3 from the zip file and it now correctly downloads the .ovpn files and appears to connect, but it just stays at "Connecting to first VPN.  Connected, checking location info."  (tried with multiple servers).  If you click 'Cancel', it says "Cancelling..." but does nothing else.  Once in this state, I can do nothing except reboot it via SSH.  I believe the logs are cleared after a reboot so they would be of no use here ?

Many thanks !

**Comment by Zomboided** [comment]
Switch on debug, wait for it to get into this state and get the kodi before you reboot. Or reboot and get the kodi.old.log from the same location

**Comment by auron476** [comment]
Hi, I'm also having the same issue and I have attached my kodi log file after switching on debug in vpn manager.

Hope this helps
[01_KODI.txt](https://github.com/Zomboided/service.vpn.manager/files/1252545/01_KODI.txt)

**Comment by Zomboided** [comment]
Thanks, this is a good log.
Did you come at this from a new install, or were you using it previously
(and if so, what version, roughly).
If you reset the VPN Provider files in settings/utilities, does this go
away?
How long did you leave it before getting the log?  It looks like it's
making the call to the IP provider, but it's not coming back (or you're not
waiting long enough for it to get upset and me to see that trace point).
Can you leave it ~15 mins and get the log?

**Comment by aneng64** [comment]
Hi. As per your recent message, here is a log file: https://pastebin.com/tWisnwvC

I have narrowed the behaviour down to ipinfo.io apparently taking forever to respond. It will eventually display the information if you wait for it to time out.

There is also another minor issue in that if you cycle through the VPN connections, the dialogue box says "Connected to VPN (say Copenhagen 06) - using profile (whatever it's supposed to be connecting to)"
In other words, they don't match.

Hope this helps.

Cheers !

**Comment by Zomboided** [comment]
This shows ipinfo is bad for you and it times out and uses the next IP
service.  Subsequent attempts to get the info come from the last service
that worked (I think it might occassionally go back to the previous one to
see if it's fixed).  If you don't want that behaviour because the first one
never works for you, then selcet the service that's working for you in the
debug menu.  Details are on the wiki if this isn't intuitive.

**Comment by aneng64** [comment]
Yep - changing to another service worked.  Thanks.  However - it does still show weird results like "Connected to a VPN in Greece, using profile Denmark - Copenhagen (06)".  What's that all about ? :-)

**Comment by Zomboided** [comment]
It's wherever the ip service thinks the IP address is.

**Comment by aneng64** [comment]
I don't think it is.  The country listed in the first part of the message (Greece in my example) is always either the primary connection in the list or the one it previously connected to (I can't remember right now), but it is definitely coming from within the plugin and not from the ip service.

**Comment by auron476** [comment]
Hi, I know that you have closed this thread but I have solved the hanging issue by going to the information display settings and changing the 'Change IP Location Service' to IP-API from ipinfo.io

The VPN manager now displays the correct vpn server information from whatever server you select.

**Comment by Zomboided** [comment]
Ok thanks. I probably need to look at the timeout and see if I can shorten it. It doesn't need to wait that long to conclude a service isn't working and it just makes the auto switching less useful

**Comment by aneng64** [comment]
It's definitely not reporting correctly regardless of ip information service you choose.  E.g. After changing VPN servers a couple of times, it's now telling me (after choosing one in Singapore) "Connected to a VPN in Capital Region, Denmark, Using profile Singapore 16 (UDP).  External address is 82.103.133.xxx.  Service provider is ASERGO Scandinavia APS".

**Comment by Zomboided** [comment]
I already told you why this was, I'm not looking at it anymore.

**Comment by aneng64** [comment]
You seem to be struggling to grasp what I'm trying to tell you, much like you obviously struggle with conversational politeness.  Thanks for the plugin anyway.
